design/session-affinity.md

```diff
@@ -0,0 +1,157 @@
+# Session Affinity
+
+Status: _Draft_
+
+Session Affinity, sometimes called _Sticky Sessions_, is a pattern whereby the incoming request indicates the backend server it wishes to handle the request.
+Sticky sessions are very much not a cloud native pattern, however when migrating applications to Kubernetes it may be required to support session affinity if the application requires them.
+
+This proposal describes how session affinity support will be added to Contour.
+
+## Goals
+
+- Support session affinity via a session cookie managed transparently by Contour.
+
+## Non Goals
+
+- Session Affinity via application supplied Header or Cookie.
+- Session Affinity via client IP.
+
+## Background
+
+Session Affinity can be enabled in Envoy via three methods; header, cookie, and client IP.
+Of these three methods, this proposal will only implement cookie based session affinity.
+
+Header based affinity is similar to cookies--cookies are headers after all--however header based routing assumes the remote client supplies this header, which further assumes it has knowledge of the application deployment.
+That is to say, for a remote client to request that a specific pod in a deployment handle its requests, that requires a priori knowledge of that pods existence.
```

Comment:
nit: s/priori/prior/g

Reply:
further nit: or 'a priori' should be italic, since it's a Latin phrase.

Reply:
wow TIL. I'm so here for the corrective punctuation advice!